lib/ansible/modules/network/ios/ios_vlan.py

```diff
@@ -211,36 +211,46 @@ def map_params_to_obj(module):
     return obj
 
 
+def parse_to_logical_rows(out):
+    started_yielding = False
+    cur_row = []
+    for l in out.splitlines()[2:]:
+        if not l:
+            """Skip empty lines."""
+            continue
+        if '0' < l[0] < '9':
+            """Line starting with a number."""
+            if started_yielding:
+                yield cur_row
+                cur_row = []  # Reset it to hold a next chunk
+            started_yielding = True
+        cur_row.append(l)
+
+    # Return the rest of it:
+    yield cur_row
+
+
+def map_ports_str_to_list(ports_str):
+    return list(filter(bool, (p.strip().replace('Gi', 'GigabitEthernet') for p in ports_str.split(', '))))
+
+
+def parse_to_obj(logical_rows):
+    first_row = logical_rows[0]
+    rest_rows = logical_rows[1:]
+    obj = re.match(r'(?P<vlan_id>\d+)\s+(?P<name>[^\s]+)\s+(?P<state>[^\s]+)\s+(?P<interfaces>.*)', first_row).groupdict()
```

Comment:
Your regex isn't properly capturing the case where there are no interfaces and also no trailing space at the end of the definition. Consider ending with `\s*(?P<interfaces>.*)`

Reply:
Good catch!

Reply:
@pierremahot we'll need a test for this